Don't let applications pass in partial paths as the service name.  Probably
we should relax this to allow subdirectories of pam.d to be accessed, but
for now leave it at the paranoid level.

diff -uNr Linux-PAM-0.75/libpam/pam_start.c Linux-PAM-0.75/libpam/pam_start.c
--- Linux-PAM-0.75/libpam/pam_start.c	Mon Jan 22 01:07:29 2001
+++ Linux-PAM-0.75/libpam/pam_start.c	Tue Jul 10 00:14:23 2001
@@ -29,6 +29,9 @@
 	return (PAM_BUF_ERR);
     }
 
+    if (strrchr(service_name, '/') != NULL)
+	service_name = strrchr(service_name, '/') + 1;
+    
     /* Mark the caller as the application - permission to do certain
        things is limited to a module or an application */
 
